gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/BloodPressureMapper.java

```diff
@@ -0,0 +1,237 @@
+            .filter(ehrComposition -> ehrComposition.getComponent()
+                .stream()
+                .map(RCMRMT030101UK04Component4::getCompoundStatement)
+                .anyMatch(Objects::nonNull))
+            .toList();
+    }
+
+    private boolean containsValidBloodPressureTriple(RCMRMT030101UK04CompoundStatement compoundStatement) {
+        var observationStatements = getObservationStatementsFromCompoundStatement(compoundStatement);
+
+        if (observationStatements.size() == 2) {
+            return BloodPressureValidatorUtil.validateBloodPressureTriple(compoundStatement.getCode().getCode(),
+                observationStatements.get(0).getCode().getCode(), observationStatements.get(1).getCode().getCode());
+        }
+
+        return false;
+    }
+
+    private II getEhrCompositionId(List<RCMRMT030101UK04EhrComposition> ehrCompositions,
+        RCMRMT030101UK04CompoundStatement compoundStatement) {
+        return ehrCompositions
+            .stream()
+            .filter(e -> e.getComponent()
+                .stream()
+                .anyMatch(f -> compoundStatement.equals(f.getCompoundStatement()))
+            ).findFirst()
+            .map(RCMRMT030101UK04EhrComposition::getId)
+            .orElse(null);
+    }
+
+    private Identifier getIdentifier(String id) {
+        return new Identifier()
+            .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL (NIAD-2021)
+            .setValue(id);
+    }
+
+    private CodeableConcept getCode(CD code) {
+        return code != null ? codeableConceptMapper.mapToCodeableConcept(code) : null;
+    }
+
+    private List<ObservationComponentComponent> getComponent(List<RCMRMT030101UK04ObservationStatement> observationStatements) {
+        var components = new ArrayList<ObservationComponentComponent>();
+
+        for (RCMRMT030101UK04ObservationStatement observationStatement
+            : observationStatements) {
+            components.add(new ObservationComponentComponent()
+                .setCode(getCode(observationStatement.getCode()))
+                .setValue(getValueQuantity(observationStatement.getValue(),
+                    observationStatement.getUncertaintyCode()))
+                .setInterpretation(getInterpretation(observationStatement.getInterpretationCode()))
+                .setReferenceRange(getReferenceRange(observationStatement.getReferenceRange())));
+        }
+
+        return components;
+    }
+
+    private String getComment(List<RCMRMT030101UK04ObservationStatement> observationStatements,
+        List<RCMRMT030101UK04NarrativeStatement> narrativeStatements) {
+        var stringBuilder = new StringBuilder();
+
+        for (RCMRMT030101UK04ObservationStatement observationStatement
+            : observationStatements) {
+            var bloodPressureText = observationStatement.getPertinentInformation().stream()
+                .filter(this::pertinentInformationHasText)
```

Comment:
Not sure if you agree with this approach but you could create a method inside of perinentInformation for hasText()